src/dyngen.cpp

```diff
@@ -180,18 +254,31 @@ bool enterFileToDynGenLibrary(World *world, void *raw_callback) {
       gLibrary,
       entry->hash,
       nullptr,
-      code,
+      entry->code,
     };
     gLibrary = newNode;
   } else {
-    node->code = code;
+    // swap code
+    entry->oldCode = node->code;
+    node->code = entry->code;
     for (auto* unit = node->dynGenNodes; unit; unit = unit->next) {
-      unit->dynGenUnit->updateCode(code);
+      // protecting in case the vm already got removed b/c the synth got removed
+      if (unit->dynGenUnit->vm != nullptr) {
+        unit->dynGenUnit->vm->init(entry->code);
+      } else {
+        return false;
+      }
     }
   }
 
-  // do not continue to stage 3 by returning false
-  return false;
+  return true;
+}
+
+// runs in stage 4 (non-RT-thread)
+bool deleteOldCode(World *world, void *raw_callback) {
+  auto entry = static_cast<NewDynGenLibraryEntry*>(raw_callback);
+  delete entry->oldCode;
```

Comment:
`oldCode` is an array that has been allocated with `new char[]`, so you have to use `delete[]`!
```
delete[] entry->oldCode;
```

---

In this case you happen to get away with it, but if the type has a non-trivial destructor, this will leak memory (or worse)! `delete []` makes sure that the destructor is called for every object in the array while `delete` will only attempt to destroy a single object.